crates/fmt/testdata/FunctionDefinition/all-params.fmt.sol

```diff
@@ -0,0 +1,736 @@
+// config: line_length = 60
+// config: multiline_func_header = "all_params"
+interface FunctionInterfaces {
+    function noParamsNoModifiersNoReturns();
+
+    function oneParam(
+        uint256 x
+    );
```

Comment:
@grandizzy I had a chance to test this today, and I see that this doesn't match the previous "multiline_func_header = all" behaviour :( 

I apologise for the confusion, but the behaviour I'm trying to recreate is the following:
- Multiline **everything** if the function definition occupies more than a single line.
- Don't multiline anything otherwise, e.g. similar to "all" for this particular case: https://github.com/grandizzy/foundry/blob/93aaebb516a7f1d52f60b07bc2f407c68198185e/crates/fmt/testdata/FunctionDefinition/all.fmt.sol#L6

Reply:
Hm, I am at a lost here, in the issue you created you mention that
```Solidity
function a(uint256 param1)
```
 it's a bug and should be expected to be as
```Solidity
function a(
    uint256 param1
)
```
which the change from this PR is doing, but actually it shouldn't be multilined now?
Please add relevant case expected vs actual and will look into, a diff would also help

Reply:
@grandizzy Here's the relevant test case:

```solidity
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.20;

contract A {
    // Single parameter is printed on a new line, when the function definition is multi-line
    function a(
        uint256 param1
    )
        external
        pure
        returns (uint256 firstReturnValue, uint256 secondReturnValue, uint256 thirdReturnValue)
    {
        firstReturnValue = param1 + 1;
        secondReturnValue = param1 + 2;
        thirdReturnValue = param1 + 3;
    }

    // Single parameter is printed on the same line, when the function definition is single-line
    function b(uint256 param1) external pure returns (uint256) {
        return param1 + 1;
    }
}
```

I think I got this working in my local foundry fork, do you mind if try opening a feature PR for that? Assuming it would be fine to modify the newly added `AllParams` option.

Reply:
yep, sure, please do a PR, `AllParams` was added to accommodate those diffs so all good. Thanks!

Reply:
For reference, I am trying to replicate the `multiline_func_header=all` behaviour from the August build we were using. The set of formatting rules made sense at the time, and still does :)